src/test/Delegation.t.sol

```diff
@@ -520,11 +583,16 @@ contract DelegationTests is EigenLayerTestHelper {
         cheats.assume(eigenAmount >= 1 && eigenAmount <= 1e18);
 
         if (!delegation.isOperator(operator)) {
-            _testRegisterAsOperator(operator, IDelegationTerms(operator));
+            IDelegationManager.OperatorDetails memory operatorDetails = IDelegationManager.OperatorDetails({
+                earningsReceiver: operator,
+                delegationApprover: address(0),
+                stakerOptOutWindowBlocks: 0
+            });
+            _testRegisterAsOperator(operator, operatorDetails);
         }
 
         //making additional deposits to the strategies
-        assertTrue(delegation.isNotDelegated(staker) == true, "testDelegation: staker is not delegate");
+        assertTrue(!delegation.isDelegated(staker) == true, "testDelegation: staker is not delegate");
```

Comment:
pls add tests for 
- force undelegation
- calling functions that are permissioned to only the strategyManager
- changing opt out blocks
- chainging metadata uri

Reply:
>     * calling functions that are permissioned to only the strategyManager

This is covered by `testCannotCallUndelegateFromNonStrategyManagerAddress`, `testCannotCallIncreaseDelegatedSharesFromNonStrategyManagerAddress`, and `testCannotCallDecreaseDelegatedSharesFromNonStrategyManagerAddress`

>     * changing opt out blocks

This is covered by `testModifyOperatorParameters`

>     * chainging metadata uri

This is covered by `testUpdateOperatorMetadataURI`

>     * force undelegation

This appears to be missing coverage; I'll add a test!

Reply:
> This appears to be missing coverage; I'll add a test!

I added the missing coverage (and refined the contract spec + behavior) in https://github.com/Layr-Labs/eigenlayer-contracts/pull/97/commits/e27672c12afcb8cbb3dd228a6bd6e0c9bf0aaddc
@gpsanant  plz review this commit!